backend/uclapi/timetable/app_helpers.py

```diff
@@ -0,0 +1,126 @@
+from roombookings.helpers import PrettyJsonResponse as JsonResponse
+from django.core.exceptions import ObjectDoesNotExist
+from .models import Stumodules, Timetable, Module, Weekstructure, \
+    Weekmapnumeric, Lecturer, Rooms, Sites
+from django.conf import settings
+import datetime
+
+_SETID = settings.ROOMBOOKINGS_SETID
+
+week_map = {}
+week_num_date_map = {}
+
+
+def get_timetable(student):
+    # To avoid hitting database multiple times.
+    if not week_map:
```

Comment:
This repetition looks incredibly ugly :( As `week_map` is global (another deadly sin), should not it be enough to do it when starting the API?

Reply:
hm I noted it. but I prefer it to be global though :/